helix-term/src/commands/lsp.rs

```diff
@@ -61,10 +61,33 @@ macro_rules! language_server_with_feature {
     }};
 }
 
+/// A wrapper around `lsp::Location` that swaps out the LSP URI for `helix_core::Uri`.
+#[derive(Debug, Clone, PartialEq, Eq)]
+struct Location {
+    uri: Uri,
+    range: lsp::Range,
+}
+
+fn lsp_location_to_location(location: lsp::Location) -> Option<Location> {
+    let uri = match location.uri.try_into() {
+        Ok(uri) => uri,
+        Err(err) => {
+            log::warn!(
```

Comment:
I think this one could use a better error message.

For example `"discarding invalid or unsupported URI {uri}: {err}"`.

The current message is very focused on the implementation but what matters to users is that `foo:://` uris are not supported by helix.